app/services/expression_evaluators/measure_unit.rb

```diff
@@ -49,11 +55,20 @@ def duty_amount_in_eur?
       component.monetary_unit_code == 'EUR'
     end
 
-    def query
-      {
-        'as_of' => user_session.import_date.iso8601,
-        'filter[geographical_area_id]' => user_session.country_of_origin,
-      }
+    def convoluted_hashes
+      @convoluted_hashes ||=
+        uk_measure_amounts.merge(xi_measure_amounts) do |_key, uk_amounts, xi_amounts|
+          uk_amounts['measure_sids'] += xi_amounts['measure_sids']
+          uk_amounts
+        end
+    end
+
+    def uk_measure_amounts
+      @uk_measure_amounts ||= filtered_commodity(source: 'uk').applicable_measure_units
+    end
+
+    def xi_measure_amounts
+      @xi_measure_amounts ||= filtered_commodity(source: 'uk').applicable_measure_units
```

Comment:
This needs to be xi here
```suggestion
      @xi_measure_amounts ||= filtered_commodity(source: 'xi').applicable_measure_units
```